packages/core/src/App/Components/Layout/Header/toggle-menu-drawer.jsx

```diff
@@ -170,7 +171,14 @@ const ToggleMenuDrawer = React.forwardRef(
         }, [is_appstore, is_pre_appstore, account_status, should_allow_authentication]);
 
         const toggleDrawer = React.useCallback(() => {
-            setIsOpen(!is_open);
+            if (!is_open) setIsOpen(!is_open);
+            else {
+                setTransitionExit(true);
+                setTimeout(() => {
```

Comment:
```suggestion
const timout = useRef();
const toggleDrawer = useCallback(() => {
timout.current = setTimout(....)
})
```
and then clean it up on component unmount phase, what do you think?